### PR TITLE
`chia wallet show` output now displays the DID ID for NFT/DID wallets

### PR DESCRIPTION
**Pull request by paninaro**
Added `nft_get_wallet_did` RPC to get the DID ID for an NFT wallet
Bech32m encode the DID ID output from `nft_get_wallets_with_dids` RPC

### CODE REVIEW

File: chia/rpc/wallet_rpc_api.py [hunk 1]
```diff
@@ -1424,7 +1439,8 @@ async def nft_get_wallets_with_dids(self, request) -> Dict:
                         did_nft_wallets.append(
                             {
                                 "wallet_id": wallet.id(),
-                                "did_id": nft_wallet_did.hex(),
+                                # "did_id": nft_wallet_did.hex(),
```
Comment (AmineKhaldi) [inline chia/rpc/wallet_rpc_api.py:1442]: Leftover?

File: chia/rpc/wallet_rpc_api.py [hunk 2]
```diff
@@ -1404,6 +1405,20 @@ async def nft_get_by_did(self, request) -> Dict:
                 return {"wallet_id": wallet.wallet_id, "success": True}
         return {"error": f"Cannot find a NFT wallet DID = {did_id}", "success": False}
 
+    async def nft_get_wallet_did(self, request) -> Dict:
+        wallet_id = uint32(request["wallet_id"])
+        assert self.service.wallet_state_manager is not None
+        nft_wallet: NFTWallet = self.service.wallet_state_manager.wallets[wallet_id]
+        if nft_wallet is not None:
+            if type(nft_wallet) != NFTWallet:
```
Comment (AmineKhaldi) [inline chia/rpc/wallet_rpc_api.py:1413]: type() compared to the uint8 of this enum member?

**Review by AmineKhaldi — DISMISSED** [review]
Looks good to me, thanks!

File: chia/rpc/wallet_rpc_api.py (hunk 2 above)
Comment (paninaro) [reply chia/rpc/wallet_rpc_api.py:1413]: NFTWallet doesn't expose a WalletType property that I see. There's a classmethod for returning the WalletType, but it doesn't apply to instances.

Comment (paninaro) [reply chia/rpc/wallet_rpc_api.py]: Nevermind, I was mistaken about how @classmethod works. Fixed now.

Comment (trepca) [reply chia/rpc/wallet_rpc_api.py]: you could use `type()` here: https://github.com/Chia-Network/chia-blockchain/blob/release/1.4.0/chia/wallet/nft_wallet/nft_wallet.py#L128

`nft_wallet.type() == WalletType.NFT`

**Review by trepca — APPROVED** [review]

**Pull request merged**